src/interfaces/vault.ts

```diff
@@ -688,4 +680,27 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     };
     return apy;
   }
+
+  async populateTransaction({
```

Comment:
should remain private

Reply:
If private, then how can we use it in the simulation class?

Reply:
might be a bit misleading of a name, since we already have `populateTransaction`? Maybe `populateDepositTransaction`

Reply:
> might be a bit misleading of a name, since we already have `populateTransaction`? Maybe `populateDepositTransaction`

Definitely, renamed with https://github.com/yearn/yearn-sdk/pull/280/commits/5ea0cc7046ed7a18f3e71e591aa16811511cf1f0